docs/src/main/tut/add-support-for-complex-types.md

```diff
@@ -0,0 +1,246 @@
+PureConfig in how to read sealed families of case classes, it prevents PureConfig from doing its 
+magic. For instance, it prevents PureConfig from using the `CorpoductHint` as strategy to read 
+`Identifiable`. This is, generally speaking, not a good idea. In theory you want PureConfig 
+features also for your custom class.
+
+##### Method 2: Use a Scala copy of the configuration
+
+PureConfig works out-of-the-box with Scala sealed familes of case classes and one simple solution
+to support unsupported complex types is to
+
+1. define a hierarchy of Scala classes mirroring the complex type
+2. define a conversion from this Scala configuration to the complex type
+
+In the case above, we could do
+
+```tut:silent:book
+// define the "dummy" scala hierarchy
+sealed trait IdentifiableDummy {
+  def id: String
+  def toIdentifiable: Identifiable
+}
+
+final case class Class1Dummy(id: String) extends IdentifiableDummy {
+  def toIdentifiable: Identifiable = new Class1(id)
+}
+
+final case class Class2Dummy(id: String, value: Int) extends IdentifiableDummy {
+  def toIdentifiable: Identifiable = new Class2(id, value) 
+}
+
+// change the coproduct hint to tell PureConfig that the type value specified
+// is the name of the class - "Dummy" because we want the type to be the one
+// of the new types, not in the mirroring types
+implicit val identifiableCoproductHint = new FieldCoproductHint[IdentifiableDummy]("type") {
+  override protected def fieldValue(name: String): String =
+     name.take(name.length - "Dummy".length).toLowerCase
+}
+
+// we tell PureConfig that to read Identifiable, it has to read IdentifiableDummy first
+// and then map it to Identifiable
+implicit val identifiableConfigReader: ConfigReader[Identifiable] =
+  ConfigReader[IdentifiableDummy].map(_.toIdentifiable)
+```
+
+**pros**: it's idiomatic and elegant. It uses a type supported by PureConfig to extract the data
+needed from the configuration and then map it to the original complex type. Note that this
+method follows what you usually do in other libraries (e.g. Circe) to support new datatypes and
+it's both composable and nice to read
+**cons**: this method needs to duplicate the configuration and PureConfig needs to do two
+steps to read it.
+
+##### Method 3: Give a Generic representation of your configuration
+
+PureConfig is based on Shapeless Generics which is used to represent data structures in a
+generic form. The reason why in method 2 PureConfig was able to decode
+`IdentifiableDummy` but not `Identifiable` prior our code is that `IdentifiableDummy`
+has a Generic instance provided by Shapeless while `Identifiable` doesn't.
+This method consists in adding the generic representation for the configuration.
+
+Let's start with the concrete classes `Class1` and `Class2`. PureConfig needs to know
+two things in order to read them:
+
+1. Their generic representation
+2. For each field in the generic representation, if there is a default value for that field or not.
```

Comment:
Suggestion:
"Whether each field in the generic representation has a default value."